persistence-spi/src/main/java/org/apache/unomi/persistence/spi/PersistenceService.java

```diff
@@ -175,6 +176,17 @@
      */
     boolean update(Item item, Date dateHint, Class<?> clazz, Map<?, ?> source, final boolean alwaysOverwrite);
 
+    /**
+     * Updates Map of items of the specified class and identified by the specified identifier with a new property value for the specified property name. Same as
+     * {@code update(itemId, dateHint, clazz, Collections.singletonMap(propertyName, propertyValue))}
+     *
+     * @param items         A map the consist of item (key) and properties to update (value)
+     * @param dateHint      a Date helping in identifying where the item is located
+     * @param clazz         the Item subclass of the item to update
+     * @return List of failed Items Ids, if all succesful then returns an empty list. if the whole operation failed then will return null
+     */
+    List<String> update(Map<Item, Map> items, Date dateHint, Class clazz);
```

Comment:
It's strange to have a Map<Item, ...> this mean that it could contains items of different types, and it may not correspond to the Class clazz parameter.
Could we implement a check before persisting the item that the item is correctly of type clazz ?

Reply:
i wonder where to have it and how.
the implementation of this method does  use the class for every item request in the bulk request.
also regular update uses a class and item without checks, they rely that it will on elastic if index is not found.

https://github.com/apache/unomi/pull/237/files#diff-cb0ba8419f934aeaa557cb1eded52b588ae98b8f1b9dbe6abd1d67553a0b27b2R977

https://github.com/apache/unomi/pull/237/files#diff-cb0ba8419f934aeaa557cb1eded52b588ae98b8f1b9dbe6abd1d67553a0b27b2R949

Reply:
Yes you are right even the regular single update is doing that, but I think it would be great to do the same check.

before processing the item we could check for:
`item.getClass().isAssignableFrom(clazz.class)`
if not
`throw new IllegalArgumentException();`

But it could be done and check in a separate PR/unomi ticket, as it is not the current PR topic.